### PR TITLE
 after install have a xqerl greeter #48 

### PR DESCRIPTION
**Pull request by grantmacken**
After user installs xqerl and runs xqerl as a server listening on port 8081.
Then user can open http://localhost:8081 and get the welcome page.

 - [x] addedsupervisor and server code for loading  xquery main modules in priv dir
 - [x] added main module 'greeter.xq` to priv/modules dir
 - [x] added associated css and image files to priv/static/assets dir 
 - [x] altered cowboy routes so `http://localhost:8081` will display greeter page 
 - [x] wrote check  in actions to show greeter is working

### CODE REVIEW

File: priv/modules/greeter.xq [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      attribute href { "/assets/images/favicon.ico"},
+      attribute sizes {"any"}
+    }
+  },
+  element body{ 
+    element img { 
+     attribute src { "/assets/images/logo_m.png" },
+     attribute alt { "xqerl logo" }
+     },
+    element h1 { 'You Are Now Flying Xqerl'  },
```
Comment (zadean) [inline priv/modules/greeter.xq:20]: ❤️

File: src/handler_greeter.erl [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+-module(handler_greeter).
```
Comment (zadean) [inline src/handler_greeter.erl:2]: I think the name of this file and module is better off being prefixed with "xqerl_". Reason being is the flat namespace of Erlang. There is a higher chance when someone is embedding xqerl into their own project that this name here could cause a conflict.

File: src/main_mod_server.erl [hunk 3]
```diff
@@ -0,0 +1,49 @@
+-module(main_mod_server).
```
Comment (zadean) [inline src/main_mod_server.erl:1]: Same thing here with a prefix to avoid namespace clash.

File: src/main_mod_sup.erl [hunk 4]
```diff
@@ -0,0 +1,45 @@
+
+-module(main_mod_sup).
```
Comment (zadean) [inline src/main_mod_sup.erl:2]: and here too with the namespace thing 😄

**Review by zadean — CHANGES_REQUESTED** [review]
Very cool!

One thing I've always strived for is to be able to embed xqerl into other Erlang or Elixir projects. I've noted a few module naming things here that could possibly cause conflicts with that. But other than that, really nice!! Thank you for all the work here!

File: src/main_mod_server.erl (hunk 3 above)
Comment (grantmacken) [reply src/main_mod_server.erl]: OK,  prefixed files and code reference with xqerl

File: src/main_mod_sup.erl (hunk 4 above)
Comment (grantmacken) [reply src/main_mod_sup.erl]: OK, prefixed files and code reference with xqerl

File: priv/modules/greeter.xq (hunk 1 above)
Comment (grantmacken) [reply priv/modules/greeter.xq:20]: NowFlyingXqerl

Comment (grantmacken) [reply priv/modules/greeter.xq:20]: Most probably the png image background color needs to be transparent

File: src/handler_greeter.erl (hunk 2 above)
Comment (grantmacken) [reply src/handler_greeter.erl]: OK.  everything should be renamed in latest push

**Comment by grantmacken** [comment]
Not sure why .github/workflows/xqerl.yml is creating a merge conflict?
I renamed it main.xqerl  reflecting the fact it should run,
 only on a push to the main branch or a pushing a tag

**Comment by grantmacken** [comment]
Close #49 for now.  Work will continue on issue #48 in newer branch.

**Pull request closed without merging**